code/compute_fc/compute_fc.py

```diff
@@ -0,0 +1,884 @@
+        description="""Compute functional connectivity matrices from fmriprep
+                    output directory.""",
+    )
+
+    # Input/Output arguments and options
+    parser.add_argument("data_dir", help="BIDS dataset or derivatives with data")
+    parser.add_argument(
+        "-s", "--save", action="store_true", default=False, help="save the outputs"
+    )
+    parser.add_argument(
+        "-o", "--output", default=None, help="specify an alternative output directory"
+    )
+
+    # Script specific options
+    parser.add_argument(
+        "--overwrite", default=False, action="store_true", help="force computation"
+    )
+    parser.add_argument(
+        "--task",
+        default=[],
+        action="store",
+        nargs="+",
+        help="a space delimited list of task(s)",
+    )
+
+    # fMRI and denoising specific options
+    # parser.add_argument(
+    #     "--denoise-only", default=False, action="store_true", help="NOT IMPLEMENTED"
+    # )
+    parser.add_argument(
+        "--atlas-dimension",
+        default=64,
+        type=int,
+        help="dimension of the atlas (usually 64, 128 or 512)",
+    )
+    parser.add_argument(
+        "--low-pass",
+        default=0.15,
+        action="store",
+        type=float,
+        help="cutoff frequency of low pass filtering",
+    )
+    parser.add_argument(
+        "--FD-thresh",
+        default=0.4,
+        action="store",
+        type=float,
+        help="framewise displacement threshold (in mm)",
+    )
+    parser.add_argument(
+        "--SDVARS-thresh",
+        default=5,
+        action="store",
+        type=float,
+        help="standardised DVAR threshold",
+    )
+    parser.add_argument(
+        "--n-scrub-frames",
+        default=5,
+        action="store",
+        type=int,
+        help="minimum segment length after volume censoring",
+    )
+    parser.add_argument(
+        "--fc-estimator",
+        default="sparse inverse covariance",
+        action="store",
+        type=str,
+        help="""type of connectivity to compute (can be 'correlation', 'covariance' or
+        'sparse')""",
+    )
+    parser.add_argument(
+        "--no-censor",
+        default=False,
+        action="store_true",
+        help="interpolate volumes with high motion without censoring",
+    )
+
+    parser.add_argument(
+        "-v",
+        "--verbosity",
+        action="count",
+        default=1,
+        help="""increase output verbosity (-v: standard logging infos; -vv: logging
+        infos and NiLearn verbose; -vvv: debug)""",
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def get_func_filenames_bids(paths_to_func_dir, task_filter=[]):
+    logging.debug("Using BIDS to find functional files...")
+
+    layout = BIDSLayout(
+        paths_to_func_dir,
+        validate=False,
+    )
+
+    all_derivatives = layout.get(
+        scope="all",
+        return_type="file",
+        extension="nii.gz",
```

Comment:
I don't remember, should this have an initial period?
```suggestion
        extension=".nii.gz",
```

Reply:
It worked well without, but maybe we could allow for uncompressed nifti files:
```suggestion
        extension=["nii.gz", "nii"],
```

Reply:
Yes, that'd be safer. Regarding the period, I don't know what is deprecated but one of the two certainly is.

Reply:
In [this example](https://bids-standard.github.io/pybids/layout/index.html#extracting-metadata) of pybids, they use `"nii.gz"`, I'll do some more tests as soon as I can access the data.